### ISSUE TITLE
Update-HPOVServerProfile : Cannot validate argument on parameter 'body'.

### ISSUE THREAD

**Issue by DerFireWal** [HewlettPackard/POSH-HPOneView#340]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
Update / Reapply of server profile as it is successfully possible using OneView Web UI.

### Actual Behavior

```PowerShell
Update-HPOVServerProfile : Cannot validate argument on parameter 'body'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
At line:1 char:91
+ ...  "Critical"} | Update-HPOVServerProfile -Async -Reapply
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Update-HPOVServerProfile], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Update-HPOVServerProfile
```

### Steps to reproduce
Get list of Server Profiles that are in "critical / locked" state and parse the to Update-HPOVServerProfile:
```PowerShell
Get-HPOVServerProfile | where {$_.Status -eq "Critical"} | Update-HPOVServerProfile -Async -Reapply
```
Profiles are correctly fetched:
```PowerShell
Get-HPOVServerProfile -ApplianceConnection $con_prod | where {$_.Status -eq "Critical"}

Name      Status   Compliance Template           Server Hardware Server Hardware Type Enclosure Group Affinity
----      ------   ---------- --------           --------------- -------------------- --------------- --------
SDZ088007 Critical Compliant  SRV-DMZ-GP_KVM-V01 S09R-E1, bay 4  BL460c_G9-FL1_536FLB EG-DMZ-v01      Bay     
SDZ088008 Critical Compliant  SRV-DMZ-GP_KVM-V01 S09R-E1, bay 5  BL460c_G9-FL1_536FLB EG-DMZ-v01      Bay     
SDZ218006 Critical Compliant  SRV-DMZ-GP_KVM-V01 S09R-E1, bay 3  BL460c_G9-FL1_536FLB EG-DMZ-v01      Bay     
SDZ218007 Critical Compliant  SRV-DMZ-GP_KVM-V01 S09R-E1, bay 6  BL460c_G9-FL1_536FLB EG-DMZ-v01      Bay     
```
Additionally tried that this way:
```PowerShell
$ProfilesToUpdate = Get-HPOVServerProfile -ApplianceConnection $con_prod | where {$_.Status -eq "Critical"}
$ProfilesToUpdate | ForEach-Object {Update-HPOVServerProfile -Reapply -InputObject $_}
Update-HPOVServerProfile : Cannot validate argument on parameter 'body'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
At line:1 char:37
+ $ProfilesToUpdate | ForEach-Object {Update-HPOVServerProfile -Reapply -InputObje ...
+                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Update-HPOVServerProfile], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Update-HPOVServerProfile
``` 

Verbose output:

```PowerShell
Get-HPOVServerProfile -ApplianceConnection $con_prod | where {$_.Status -eq "Critical"} | Update-HPOVServerProfile -Async -Reapply -verbose
VERBOSE: [UPDATE-HPOVSERVERPROFILE] Bound PS Parameters: 
Key                                                                                             Value                                                                                         
---                                                                                             -----                                                                                         
Async                                                                                           True                                                                                          
Reapply                                                                                         True                                                                                          
Verbose                                                                                         True                                                                                          



VERBOSE: [UPDATE-HPOVSERVERPROFILE] Called from: <ScriptBlock>
VERBOSE: [UPDATE-HPOVSERVERPROFILE] Processing Pipeline input.
VERBOSE: [UPDATE-HPOVSERVERPROFILE] Processing Server Profile: 'SDZ088007 [/rest/server-profiles/eb19eb3d-7388-44fd-9580-53afaad9ec76]'
VERBOSE: [SEND-HPOVREQUEST] BEGIN
VERBOSE: [SEND-HPOVREQUEST] Bound PS Parameters: 
Key                                                                                             Value                                                                                         
---                                                                                             -----                                                                                         
uri                                                                                             /rest/server-hardware/31333138-3839-5A43-3337-333854333732                                    
Hostname                                                                                        sdit11122.duit.loc                                                                            



VERBOSE: [SEND-HPOVREQUEST] Process
VERBOSE: [SEND-HPOVREQUEST] Hostname value: 
Name                                                                                                                                                                              ConnectionId
----                                                                                                                                                                              ------------
sdit11122.duit.loc                                                                                                                                                                           1



VERBOSE: [SEND-HPOVREQUEST] Filtering for Connection Object via PSObject: 
Name                                                                                                                                                                              ConnectionId
----                                                                                                                                                                              ------------
sdit11122.duit.loc                                                                                                                                                                           1



VERBOSE: [SEND-HPOVREQUEST] Processing 'sdit11122.duit.loc' appliance connection request. 1 of 1
VERBOSE: [SEND-HPOVREQUEST] Requested URI '/rest/server-hardware/31333138-3839-5A43-3337-333854333732' to 'sdit11122.duit.loc'
VERBOSE: [RESTCLIENT] Building new [System.Net.HttpWebRequest] object for GET https://sdit11122.duit.loc/rest/server-hardware/31333138-3839-5A43-3337-333854333732
VERBOSE: [SEND-HPOVREQUEST] Request: GET https://sdit11122.duit.loc/rest/server-hardware/31333138-3839-5A43-3337-333854333732
VERBOSE: [SEND-HPOVREQUEST] Request Header 1: User-Agent = HPEOneview_PowerShell/4.00
VERBOSE: [SEND-HPOVREQUEST] Request Header 2: Content-Type = application/json
VERBOSE: [SEND-HPOVREQUEST] Request Header 3: Accept = application/json
VERBOSE: [SEND-HPOVREQUEST] Request Header 4: X-API-Version = 600
VERBOSE: [SEND-HPOVREQUEST] Request Header 5: accept-language = en_US
VERBOSE: [SEND-HPOVREQUEST] Request Header 6: accept-encoding = gzip, deflate
VERBOSE: [SEND-HPOVREQUEST] Request Header 7: auth = [*****REDACTED******]
VERBOSE: [SEND-HPOVREQUEST] Response Status: 200 (OK)
VERBOSE: [SEND-HPOVREQUEST] Response Header 1: Transfer-Encoding = chunked
VERBOSE: [SEND-HPOVREQUEST] Response Header 2: Cache-Control = no-cache
VERBOSE: [SEND-HPOVREQUEST] Response Header 3: Content-Type = application/json;charset=UTF-8
VERBOSE: [SEND-HPOVREQUEST] Response Header 4: Date = Fri, 01 Jun 2018 13:13:54 GMT
VERBOSE: [SEND-HPOVREQUEST] Response Header 5: ETag = 1527698347875
VERBOSE: [SEND-HPOVREQUEST] Response Header 6: Server = Apache
VERBOSE: [SEND-HPOVREQUEST] FinalResponse: {"type":"server-hardware-8","name":"S09R-E1, bay 4","serverName":"","state":"ProfileError","stateReason":"NotApplicable","assetTag":"","category":"s
erver-hardware","created":"2018-04-16T11:24:45.736Z","description":null,"eTag":"1527698347875","formFactor":"HalfHeight","hostOsType":null,"intelligentProvisioningVersion":"Unknown","licensin
gIntent":"OneViewNoiLO","locationUri":"/rest/enclosures/09CZ3219HWW6","memoryMb":524288,"migrationState":"NotApplicable","model":"ProLiant BL460c Gen9","modified":"2018-05-30T16:39:07.875Z","
mpFirmwareVersion":"2.55 Aug 16 2017","mpHostInfo":{"mpHostName":"ILOCZ3738T372","mpIpAddresses":[{"address":"172.26.67.68","type":"Undefined"}]},"mpModel":"iLO4","mpState":"OK","partNumber":
"813198R-B21","physicalServerHardwareUri":null,"portMap":{"deviceSlots":[{"deviceName":"HP FlexFabric 10Gb 2-port 536FLB Adapter","deviceNumber":9,"location":"Flb","physicalPorts":[{"intercon
nectPort":4,"interconnectUri":"/rest/interconnects/663d7ab4-2614-4188-877e-ae621890ac4f","mac":"6C:C2:17:3E:27:F0","physicalInterconnectPort":4,"physicalInterconnectUri":"/rest/interconnects/
663d7ab4-2614-4188-877e-ae621890ac4f","portNumber":1,"type":"Ethernet","virtualPorts":[{"currentAllocatedVirtualFunctionCount":-1,"mac":"6C:C2:17:3E:27:F0","portFunction":"a","portNumber":1,"
wwnn":null,"wwpn":null},{"currentAllocatedVirtualFunctionCount":-1,"mac":"56:56:FB:80:00:04","portFunction":"b","portNumber":2,"wwnn":"10:00:02:58:60:b0:00:09","wwpn":"10:00:02:58:60:b0:00:08
"},{"currentAllocatedVirtualFunctionCount":-1,"mac":"6C:C2:17:3E:27:F2","portFunction":"c","portNumber":3,"wwnn":null,"wwpn":null},{"currentAllocatedVirtualFunctionCount":-1,"mac":"6C:C2:17:3
E:27:F3","portFunction":"d","portNumber":4,"wwnn":null,"wwpn":null}],"wwn":null},{"interconnectPort":4,"interconnectUri":"/rest/interconnects/f65a2b9b-44b6-4e6c-9363-f582d20f383c","mac":"6C:C
2:17:3E:27:F8","physicalInterconnectPort":4,"physicalInterconnectUri":"/rest/interconnects/f65a2b9b-44b6-4e6c-9363-f582d20f383c","portNumber":2,"type":"Ethernet","virtualPorts":[{"currentAllo
catedVirtualFunctionCount":-1,"mac":"6C:C2:17:3E:27:F8","portFunction":"a","portNumber":1,"wwnn":null,"wwpn":null},{"currentAllocatedVirtualFunctionCount":-1,"mac":"56:56:FB:80:00:05","portFu
nction":"b","portNumber":2,"wwnn":"10:00:02:58:60:b0:00:0b","wwpn":"10:00:02:58:60:b0:00:0a"},{"currentAllocatedVirtualFunctionCount":-1,"mac":"6C:C2:17:3E:27:FA","portFunction":"c","portNumb
er":3,"wwnn":null,"wwpn":null},{"currentAllocatedVirtualFunctionCount":-1,"mac":"6C:C2:17:3E:27:FB","portFunction":"d","portNumber":4,"wwnn":null,"wwpn":null}],"wwn":null}],"slotNumber":1},{"
deviceName":"","deviceNumber":1,"location":"Mezz","physicalPorts":[],"slotNumber":1},{"deviceName":"","deviceNumber":2,"location":"Mezz","physicalPorts":[],"slotNumber":2}]},"position":4,"pow
erLock":false,"powerState":"Off","processorCoreCount":14,"processorCount":2,"processorSpeedMhz":2400,"processorType":"Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz","refreshState":"NotRefreshing"
,"remoteSupportSettings":{"remoteSupportCurrentState":"Registered","destination":"172.26.35.48"},"remoteSupportUri":"/rest/support/server-hardware/31333138-3839-5A43-3337-333854333732","romVe
rsion":"I36 v2.30 (09/12/2016)","scopesUri":"/rest/scopes/resources/rest/server-hardware/31333138-3839-5A43-3337-333854333732","serialNumber":"CZ3738T372","serverFirmwareInventoryUri":"/rest/
server-hardware/31333138-3839-5A43-3337-333854333732/firmware","serverGroupUri":"/rest/enclosure-groups/11c3f1f3-a729-44ca-9c4c-e6e2e71c9b57","serverHardwareTypeUri":"/rest/server-hardware-ty
pes/0C27FEAA-6491-4255-B945-B628BB76B221","serverProfileUri":"/rest/server-profiles/eb19eb3d-7388-44fd-9580-53afaad9ec76","serverSettings":{"firmwareAndDriversInstallState":{"installedStateTi
mestamp":null,"installState":"Unknown"},"hpSmartUpdateToolStatus":{"mode":null,"version":null,"serviceState":null,"installState":"NotInstalled","lastOperationTime":null},"firmwareInstallSched
ule":{"dateTime":null,"scheduleOptions":"StageImmediatelyAndRetryAtDeployTime"}},"shortModel":"BL460c Gen9","signature":{"personalityChecksum":-794845687,"serverHwChecksum":-923052477},"statu
s":"Warning","supportDataCollectionState":"Completed","supportDataCollectionType":"ActiveHealthSystem","supportDataCollectionsUri":"/rest/support/data-collections?deviceID=31333138-3839-5A43-
3337-333854333732&category=server-hardware","supportState":"Enabled","uidState":"Unsupported","uri":"/rest/server-hardware/31333138-3839-5A43-3337-333854333732","uuid":"31333138-3839-5A43-333
7-333854333732","virtualSerialNumber":null,"virtualUuid":null}
VERBOSE: [SEND-HPOVREQUEST] Manual Pagination: False
VERBOSE: [SEND-HPOVREQUEST] Response object, no paging needed.
VERBOSE: [SEND-HPOVREQUEST] Cleaning up HttpWebRequest
VERBOSE: [SEND-HPOVREQUEST] Does nextPageUri member exist: False
VERBOSE: [SEND-HPOVREQUEST] Is nextPageUri Null or Empty: True
VERBOSE: [SEND-HPOVREQUEST] Stopping Do/Until loop because nextPageUri doesn't exist and have received all objects.
VERBOSE: [SEND-HPOVREQUEST] End
VERBOSE: [UPDATE-HPOVSERVERPROFILE] Sending request to Reapply configuration
Update-HPOVServerProfile : Cannot validate argument on parameter 'body'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
At line:1 char:91
+ ...  "Critical"} | Update-HPOVServerProfile -Async -Reapply -verbose
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Update-HPOVServerProfile], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Update-HPOVServerProfile


```

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion` or `$PSLibraryVersion`): 4.0.1672.1789
HPE OneView Appliance Version (`Get-HPOVVersion -ApplianceVer`):  4.00.09.345092.00
Output from `$PSVersionTable` on your Windows Host:
Name                           Value                                                                                                                                                          
----                           -----                                                                                                                                                          
PSVersion                      4.0                                                                                                                                                            
WSManStackVersion              3.0                                                                                                                                                            
SerializationVersion           1.1.0.1                                                                                                                                                        
CLRVersion                     4.0.30319.34014                                                                                                                                                
BuildVersion                   6.3.9600.18773                                                                                                                                                 
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0}                                                                                                                                           
PSRemotingProtocolVersion      2.2

**Comment by DerFireWal** [comment]
Hello Chris,

thanks for your reply - using all of the parameter switches did the trick. Didn't think about trying that.

Thanks,
Daniel

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 4.0.1821.1629](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.0.1821.1629) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.400/4.0.1821.1629).

Please verify before this issue is closed.

**Comment by ChrisLynchHPE** [comment]
Closing due to no further activity.
